### PR TITLE
530 write test cases for candidate assessment side

### PR DESCRIPTION
**Pull request by ayushicopods**
Test cases for candidate assessment

### CODE REVIEW

File: app/components/assessment/CandidateInstruction.tsx [hunk 1]
```diff
@@ -45,7 +45,7 @@ const CandidateInstruction = () => {
       <div className="flex-1 overflow-auto">
         <div className="flex flex-col items-center justify-center gap-16 py-10">
           <div className="flex flex-col items-center justify-center gap-4">
-            <h3 className="text-2xl font-bold text-gray-900">
+            <h3 className="heading text-2xl font-bold text-gray-900">
```
Comment (ayushicopods) [inline app/components/assessment/CandidateInstruction.tsx:48]: Remove `heading` from the class name. Use `data-cy` for accessing the HTML node in cypress

File: app/components/assessment/CandidateInstruction.tsx [hunk 2]
```diff
@@ -104,7 +113,10 @@ const CandidateInstruction = () => {
           <div className="flex flex-col items-center justify-center gap-4">
             <div className="flex gap-2">
               <span>🎉 </span>
-              <p className="text-base font-semibold text-gray-900">
+              <p
+                className="text-base font-semibold text-gray-900"
+                data-cy="gesture"
```
Comment (ayushicopods) [inline app/components/assessment/CandidateInstruction.tsx:118]: using intuitive text for texting purpose. like `goodLuckMessageText`

File: app/components/assessment/CandidateRegister.tsx [hunk 3]
```diff
@@ -60,6 +60,7 @@ function CandidateRegister() {
             className="h-12 w-full text-base"
             isDisabled={!canSubmitBtnBeEnabled() || busy}
             variant="primary-solid"
+            data-cy="submit-button"
```
Comment (ayushicopods) [inline app/components/assessment/CandidateRegister.tsx:63]: try to follow same case format, like if you're using camelCase using it throughout.

File: app/components/tests/SelectSectionCard.tsx [hunk 4]
```diff
@@ -74,6 +74,7 @@ const SelectSectionCard = ({
         <h3
           title={section.name}
           className="flex-1 text-xl font-semibold text-gray-700"
+          data-cy="section-heading"
```
Comment (ayushicopods) [inline app/components/tests/SelectSectionCard.tsx:77]: same as above comment.

File: app/models/assessment.server.ts [hunk 5]
```diff
@@ -99,6 +99,10 @@ export async function verifyOTP({
       candidate: { select: { OTP: true, id: true } },
     },
   })
+  if (process.env.PORT === '8811' && otp === Number('0000')) {
```
Comment (ayushicopods) [inline app/models/assessment.server.ts:102]: Could you add a comment to the code block about what it's doing and why we are doing it? For other people to understand

File: cypress/e2e/group-by-test/CandidateListOfTestItem.cy.ts [hunk 6]
```diff
@@ -23,6 +23,7 @@ describe('Visiting group by test of results page', () => {
     cy.get('#group-by-tests').should('have.text', 'Results').click()
     cy.get('h1', { timeout: 6000 }).should('have.text', 'Results')
     cy.get('.groupByItemTest', { timeout: 8000 }).contains(test1).click()
+    cy.wait(1000)
```
Comment (ayushicopods) [inline cypress/e2e/group-by-test/CandidateListOfTestItem.cy.ts:26]: review this `cy.wait()` with @anuragP04 and @richa-copods

File: cypress/e2e/group-by-test/groupByTestTable.cy.ts [hunk 7]
```diff
@@ -92,8 +92,7 @@ describe('Test for GroupByTestTable, Result', () => {
   })
   it('Total Count of Test of groupByTable', () => {
     cy.get('#group-by-tests').should('have.text', 'Results').click()
-
-    cy.get('#total-count-value', { timeout: 60000 }).should('have.text', 1)
+    cy.get('#total-count-value', { timeout: 6000 }).should('be.visible')
```
Comment (ayushicopods) [inline cypress/e2e/group-by-test/groupByTestTable.cy.ts:95]: same as above

File: cypress/e2e/smoke.cy.ts [hunk 8]
```diff
@@ -309,13 +310,13 @@ describe('smoke tests', () => {
     })
     // user reached to step 2
     cy.get('div#section').each((el) => {
+      cy.log('element', el)
```
Comment (ayushicopods) [inline cypress/e2e/smoke.cy.ts:313]: remove the log.

File: cypress/e2e/tests-candidate-assessment/A_createAssessment.cy.ts [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+    cy.get('#sections', { timeout: 6000 }).click()
+    cy.get('#add-section').click()
+    cy.get('form > div').within((el) => {
+      cy.get('input[placeholder="Enter Test Name*"]').type(
+        commonConstants?.section1
+      )
+      cy.get('textarea').type('Aptitude')
+      cy.get('[data-cy="submit"]').click()
+    })
+    cy.wait(1000)
```
Comment (ayushicopods) [inline cypress/e2e/tests-candidate-assessment/A_createAssessment.cy.ts:32]: review the `await`

File: cypress/e2e/tests-candidate-assessment/A_createAssessment.cy.ts [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+
+  it('Add question to the first section', () => {
+    cy.login()
+    cy.customVisit('/members')
+    cy.get('#sections', { timeout: 6000 }).should('have.text', 'Tests').click()
+    cy.wait(1000)
+    cy.get('.sectionName', { timeout: 6000 })
+      .contains(commonConstants?.section1)
+      .click()
+    cy.wait(1000)
```
Comment (ayushicopods) [inline cypress/e2e/tests-candidate-assessment/A_createAssessment.cy.ts:51]: same

File: cypress/e2e/tests-candidate-assessment/A_createAssessment.cy.ts [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+          ;[...el].map((el) => {
+            if (el.innerText === 'Text') {
+              el.click()
+            }
+            return null
+          })
+        })
+      })
+    })
+    cy.wait(1000)
```
Comment (ayushicopods) [inline cypress/e2e/tests-candidate-assessment/A_createAssessment.cy.ts:66]: .

**Review by ayushicopods — COMMENTED** [review]
Code reviewed by @zia-copods 

request code changes.

File: app/components/assessment/CandidateInstruction.tsx [hunk 12]
```diff
@@ -104,7 +113,10 @@ const CandidateInstruction = () => {
           <div className="flex flex-col items-center justify-center gap-4">
             <div className="flex gap-2">
               <span>🎉 </span>
-              <p className="text-base font-semibold text-gray-900">
+              <p
+                className="text-base font-semibold text-gray-900"
+                data-cy="goodLuckMessageText"
```
Comment (yash6102000) [inline app/components/assessment/CandidateInstruction.tsx:118]: don't think so we have to write test case for this

File: app/components/assessment/CandidateInstruction.tsx [hunk 13]
```diff
@@ -104,7 +113,10 @@ const CandidateInstruction = () => {
...
+              >
```
Comment (yash6102000) [inline app/components/assessment/CandidateInstruction.tsx:119]: add this in constant(index file)

File: cypress/e2e/tests-candidate-assessment/B_Registration.cy.ts [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+    cy.window()
+      .its('navigator.clipboard')
+      .invoke('readText')
+      .then((text) => {
+        examLink = text.split('3000')[1]
+        cy.visit(examLink)
+      })
+  })
+
+  it('Checks, register heading should be visible', () => {
```
Comment (yash6102000) [inline cypress/e2e/tests-candidate-assessment/B_Registration.cy.ts:31]: can we check text in same test case

File: cypress/e2e/tests-candidate-assessment/C_OtpVerification.cy.ts [hunk 15]
```diff
@@ -0,0 +1,55 @@
+describe('Test for candidate assessment verfication', () => {
+  it('registration', () => {
```
Comment (yash6102000) [inline cypress/e2e/tests-candidate-assessment/C_OtpVerification.cy.ts:2]: we don't need to create a sapeate test case for this just write it in below and add comment

File: cypress/e2e/tests-candidate-assessment/C_OtpVerification.cy.ts [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+  })
+
+  it('Checks, header of verification page should be visible and correct', () => {
+    cy.customVisitOnCandidateSide('verification')
+    cy.get('[data-cy="header"]', { timeout: 6000 })
+      .should('be.visible')
+      .should('have.text', 'OTP Verification')
+  })
+
+  it('Checks, Email Id of candidate should be visible and correct', () => {
```
Comment (yash6102000) [inline cypress/e2e/tests-candidate-assessment/C_OtpVerification.cy.ts:13]: as per your test expectation i can't see the test cases to the if email is correct or not

File: cypress/e2e/tests-candidate-assessment/D_Instructions.cy.ts [hunk 17]
```diff
@@ -0,0 +1,69 @@
+describe('Test for Assessment Verification', () => {
+  let commonContants = {
+    Begin: 'Begin Assessment',
```
Comment (yash6102000) [inline cypress/e2e/tests-candidate-assessment/D_Instructions.cy.ts:3]: for constant use camalCase

File: cypress/e2e/tests-candidate-assessment/D_Instructions.cy.ts [hunk 18]
```diff
@@ -0,0 +1,69 @@
...
+    BestWishes: 'Best of Luck',
+  }
+
+  it('Check, Assessment registration and verification', () => {
+    cy.candidateRegistration()
+    cy.candidateVerification()
+  })
+
+  it('Check, heading is visible and should have first name', () => {
```
Comment (yash6102000) [inline cypress/e2e/tests-candidate-assessment/D_Instructions.cy.ts:12]: don not required to write this this test case here

File: cypress/e2e/tests-candidate-assessment/E_TestSection.cy.ts [hunk 19]
```diff
@@ -0,0 +1,245 @@
+const commonContants = {
+  sectionHeadingOne: 'Section 1 - 2: Quantitative - assessment1',
+  sectionHeadingTwo: 'Section 2 - 2: Quantitative - assessment1',
+  coolDown: 'cooldown',
+  congratulationText: 'Congratulations! You have completed the exam',
+  Next: 'Next',
```
Comment (yash6102000) [inline cypress/e2e/tests-candidate-assessment/E_TestSection.cy.ts:6]: it should be next

File: cypress/e2e/tests-candidate-assessment/E_TestSection.cy.ts [hunk 20]
```diff
@@ -0,0 +1,245 @@
...
+  Previous: 'Previous',
```
Comment (yash6102000) [inline cypress/e2e/tests-candidate-assessment/E_TestSection.cy.ts:7]: previous

File: cypress/e2e/tests-candidate-assessment/E_TestSection.cy.ts [hunk 21]
```diff
@@ -0,0 +1,245 @@
...
+  EndTest: 'End Test',
```
Comment (yash6102000) [inline cypress/e2e/tests-candidate-assessment/E_TestSection.cy.ts:8]: same here

File: cypress/e2e/tests-candidate-assessment/E_TestSection.cy.ts [hunk 22]
```diff
@@ -0,0 +1,245 @@
...
+    cy.tick(1000)
+  })
+
+  it('Checks, Question section heading should be visible and correct', () => {
+    cy.get('[data-cy="questionSection"] span')
+      .should('be.visible')
+      .should('have.text', 'Question ')
+  })
+
+  it('Checks, Question should be visible', () => {
```
Comment (yash6102000) [inline cypress/e2e/tests-candidate-assessment/E_TestSection.cy.ts:53]: no need to create saperate test case for this please add this in another one and just add comment in the top of that

File: cypress/e2e/tests-candidate-assessment/E_TestSection.cy.ts [hunk 23]
```diff
@@ -0,0 +1,245 @@
...
+
+  it('Checks, Start new section button should be visible and works fine ', () => {
+    cy.contains('button', 'Start New Section')
+      .should('be.visible')
+      .should('have.class', 'text-gray-50')
+      .and('have.class', 'hover:bg-primaryHover')
+    cy.contains('button', 'Start New Section').click()
+  })
+
+  /**
```
Comment (yash6102000) [inline cypress/e2e/tests-candidate-assessment/E_TestSection.cy.ts:124]: can we write this comment in single line

**Review by anuragP04 — APPROVED** [review]

**Pull request merged**